src/cogent3/app/composable.py

```diff
@@ -992,9 +992,10 @@ def wrapped(klass):
             slot_attrs.remove("input")
         else:
             setattr(klass, "input", None)
-        # If a developer has defined slots, we should extend them with our own instance variables.
+
```

Comment:
`_getstate` and `_setstate`  are not used. Please remove them.

Reply:
I think we need a mechanism to prevent repeated decorating of a class.
For example:
```
if hasattr(klass, "app_type"):
    raise Type_Error ("The class {...} has app_type property. This class may have already been decorated.")
```